core/src/main/java/org/pentaho/platform/util/Emailer.java

```diff
@@ -334,7 +349,21 @@ public boolean send() {
       msg.setHeader( "X-Mailer", Emailer.MAILER ); //$NON-NLS-1$
       msg.setSentDate( new Date() );
 
-      Transport.send( msg );
+      if ( emailConfig.getAuthMechanism().equals( "XOAUTH2" ) ) {
+        if ( emailConfig.getSmtpProtocol().equalsIgnoreCase( "graph_api" ) ) {
```

Comment:
The code in this method looks similar to parts of `EmailService#sendEmail` and  `sendEmailTest`. Consider refactoring it.